test/capi/capiLinearGradient.cpp

```diff
@@ -162,7 +162,7 @@ TEST_CASE("Linear Gradient clear data", "[capiLinearGradient]")
 
     REQUIRE(tvg_gradient_set_color_stops(gradient, NULL, 0) == TVG_RESULT_SUCCESS);
     REQUIRE(tvg_gradient_get_color_stops(gradient, &color_stops_ret, &color_stops_count_ret) == TVG_RESULT_SUCCESS);
-    REQUIRE(color_stops_ret == NULL);
+    REQUIRE(color_stops_ret == (const Tvg_Color_Stop*)NULL);
```

Comment:
I guess we could achieve success by applying compiler flags. Maybe we could try using the -Wnonnull-compare flag?

Reply:
the problem here is comparison between pointer and integer, not a non-null ptr with null, so I don;t see how this flag should help here. beside, I don;t like to suppress warnings, I prefer to solve them...

Reply:
1. It would be better if you addressed the warning msg in this fix so that we can identify the issue easier. You can refer to this commit (https://github.com/thorvg/thorvg/commit/e576e1d8e6a78b97e23bbb443777a97a4ce85cfd) how I share the issue info and where the compiler notify the warning flags us.

2. Comparing NULL(0) is absolutely c-style code. "(const Tvg_Color_Stop*)NULL" this one is more alien-wise. Later some other programmers would come up with NULL compare again, we will face this issue again. 

3. Current test code is based on C++ due to the testframework, So you can just use nullptr or just shutdown warning to prevent later same issues. It's more readable and maintainable.

Reply:
the question is why there is NULL, rather than 0

Reply:
nullptr is probably better. NULL is defined from an int, while nullptr is a pointer (since C++11)

Reply:
@hermet no alien-wise code :D nullptr used

as I wrote in the pr title - it's an error, not a warning.

also, I forgot to add the issue number in the commit msg... sorry. 
I hope the issue number is enough and there is no need to put the error msg in the commit msg. Please let me know if you do not agree, so in the future I'll apply the preferred way of writing commit msg,